### PR TITLE
feat: submit resource isolation initialization framework

### PR DESCRIPTION
**Pull request by yuleichun-striving**
#### What type of PR is this?
feat: new feature
#### What does this PR do?
submit resource isolation initialization framework

#### Which issue(s) does this PR fix?
Fixes #https://github.com/kosmos-io/kosmos/issues/20

#### Screenshots (if applicable):

```release-note

```

#### Additional Comments:

```release-note

```

### CODE REVIEW

File: cmd/koslet/app/koslet.go [hunk 1]
```diff
@@ -49,57 +48,46 @@ func NewKosletCommand(ctx context.Context) *cobra.Command {
...
 
-		Callbacks: leaderelection.LeaderCallbacks{
-			OnStartedLeading: func(ctx context.Context) {
-				done = make(chan struct{})
-				defer close(done)
+	go func() {
+		if err = mgr.Start(ctx); err != nil {
+			klog.Errorf("failed to start koslet manager: %v", err)
+		}
+	}()
```
Comment (duanmengkk) [inline cmd/koslet/app/koslet.go]: No need to start a goroutine here

File: pkg/koslet/manager/controller/koslet_controller.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	},
+	UpdateFunc: func(updateEvent event.UpdateEvent) bool {
+		return true
+	},
+	DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+		return true
+	},
+	GenericFunc: func(genericEvent event.GenericEvent) bool {
+		return false
+	},
```
Comment (duanmengkk) [inline pkg/koslet/manager/controller/koslet_controller.go]: No need to use predicate func here

**Comment by wuyingjun-lucky** [comment]
> /approved

approve not approved

**Comment by wuyingjun-lucky** [comment]
@yuleichun-striving  please rebase the commits

**Comment by yuleichun-striving** [comment]
> @yuleichun-striving please rebase the commits

OK, I've taken care of it, please review

**Pull request closed without merging**